tuxemon/event/actions/transfer_money.py

```diff
@@ -15,50 +15,55 @@
 @dataclass
 class MoneyMathAction(EventAction):
     """
-    Performs a mathematical transaction on the player's money.
+    Performs a money transaction between entities.
 
```

Comment:
looking through the code it seems like there are a lot of soft-bugs or "gotchas" here.  for example, if you want to move money from the player to another entity, but the entity is not found, the player will still gain or lose the money.  this will be a huge problem since the error isn't handled.  i know the problem already existed, but could we add a comment here warning the developer that this isn't a robust system?  i don't expect you to improve it now, since the pr is just adding a different feature, but we should get a comment in here before somebody forgets.
suggestion:
The logic in handling money transactions is not covered by tests and is known to have issues.

Reply:
the logic is faulty, **transfer_money** needs to do only its task (transferring money), this is the reason why I'm going to modify **set_money** by implementing a series of addition and creating **modify_money** (new event action) that will add/remove money from the wallet

since the totality of transactions in the game is the player gaining / reducing the amount of money without someone else gaining / reducing, I have replaced the event action (not many, but better do the task now than later);

Reply:
Ok, now **transfer_money** looks clear, because it ships the different functions to precise actions.